tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/MemStore.java

```diff
@@ -579,10 +578,19 @@ public void dropPartitionMethod(String databaseName, String tableName) throws Ca
   }
 
   public List<TablePartitionProto> getAllPartitions() throws CatalogException {
+    int tableId = 0, partitionId = 0;
+    List<TableDescriptorProto> tables = getAllTables();
     List<TablePartitionProto> protos = new ArrayList<TablePartitionProto>();
-    Set<String> tables = partitions.keySet();
-    for (String table : tables) {
-      Map<String, CatalogProtos.PartitionDescProto> entryMap = partitions.get(table);
+
+    Set<String> partitionTables = partitions.keySet();
+    for (String partitionTable : partitionTables) {
+      for (TableDescriptorProto table : tables) {
```

Comment:
Since it is guaranteed that the tables corresponding to partitions always exist, it looks not necessary to match table names.